packages/server-wallet/src/engine/engine.ts

```diff
@@ -847,60 +869,88 @@ export class SingleThreadedEngine
       o => o.type === 'OpenChannel' && o.data.fundingStrategy === 'Direct'
     );
 
-    const signingAddress = await this.getCachedSigningAddress();
+    const {address} = await this.getCachedSigningWallet();
     ((direct ?? []) as OpenChannel[]).map(o => {
       const openingState = wirePayload.signedStates
         ?.map(deserializeState)
         .find(s => calculateChannelId(s) === o.data.targetChannelId);
-      this.storeRichObjective(o, openingState as State, signingAddress);
+      this.storeRichObjective(o, openingState as State, address);
     });
 
     const {signedStates} = wirePayload;
     if (signedStates?.[0]) {
-      // When funding is connected, we won't need this "backdoor exit". Instead, the objective
-      // would have already reached the 'success' state when we receive turn number 4.
-      if (signedStates[0].turnNum >= 4) return;
+      if (
+        // When funding is connected, we won't need this "backdoor exit". Instead, the objective
+        // would have already reached the 'success' state when we receive turn number 4.
+        signedStates[0].turnNum >= 4 ||
```

Comment:
In "production", when the engine receives a signed state it should:
1. check if there's an active objective for the channel.
2. If yes, send the state to the correct cranker
3. If no, send the state to the store (or later, to the AppManager)

For this prototype code, I don't want to send the event to the cranker after the channel has been funded, so this is a hacky way to avoid this.

Another way to return early in `crankRichObjective` if the objective that's stored is in the `success` state. I'll probably make that change to remove this confusing branch?

Reply:
> Another way to return early in `crankRichObjective` if the objective that's stored is in the `success` state. I'll probably make that change to remove this confusing branch?

This would be my preference.

Reply:
b2874817d29c2ab9998ebf4a41bcec3ae160c8f1